homeassistant/components/zwave_js/entity.py

```diff
@@ -61,19 +61,35 @@ def device_info(self) -> dict:
             "identifiers": {get_device_id(self.client, self.info.node)},
         }
 
-    @property
-    def name(self) -> str:
-        """Return default name from device name and value name combination."""
+    def generate_name(
+        self,
+        alternate_value_name: Optional[str] = None,
+        additional_info: Optional[List[str]] = None,
+    ) -> str:
+        """Generate entity name."""
+        if additional_info is None:
+            additional_info = []
         node_name = self.info.node.name or self.info.node.device_config.description
         value_name = (
-            self.info.primary_value.metadata.label
+            alternate_value_name
+            or self.info.primary_value.metadata.label
             or self.info.primary_value.property_key_name
             or self.info.primary_value.property_name
         )
+        name = f"{node_name}: {value_name}"
+        for item in additional_info:
+            if item:
+                name += f" - {item}"
         # append endpoint if > 1
         if self.info.primary_value.endpoint > 1:
-            value_name += f" ({self.info.primary_value.endpoint})"
-        return f"{node_name}: {value_name}"
+            name += f" ({self.info.primary_value.endpoint})"
+
+        return name
+
+    @property
+    def name(self) -> str:
+        """Return default name from device name and value name combination."""
+        return self.generate_name()
```

Comment:
Can we cache the name in an instance attribute?

Reply:
good idea

Reply:
Is that smart with all these dynamic values ?

Reply:
we have to assume everywhere that values are lazy

Reply:
If it doesn't work, we shouldn't do it. You know better than me about this.

Reply:
I'm not sure at what points we actually check the name property, but the name shouldn't be jumping around through the lifetime of the HA instance IMO. It's fine to regenerate at startup, but once that has been generated, I think it makes sense to keep it that way

Reply:
If the user hasn't set a name in the entity registry the name from the `name` property will be used when setting state.

Reply:
That's a fair point. We have to draw a line somewhere. We have at least ensured ourselves for the -as-complet-as-possible state by waiting for node ready before creating entities

Reply:
In that case I think we use this newly suggested behavior. It's not great that a name can change between HA restarts either, but that can't be helped due to the lazy nature of obtaining values